### DISCUSSION TITLE
`missing xcrun` error while compiling `imessage-exporter`

### DISCUSSION THREAD

**Discussion by bhattmayank** [ReagentX/imessage-exporter#326] [Q&A] [answered]
```
= note: xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun

error: could not compile `typenum` (build script) due to 1 previous error
error: failed to compile `imessage-exporter v2.0.1`, intermediate artifacts can be found at `/var/folders/p1/hwh7m0790xv9r0dc_0p_y6f40000gn/T/cargo-installbov487`.
```

Any idea how to fix this?
many thanks.

**Comment by bhattmayank** [comment, marked as answer]
It was issue with my environment - not bug as I am closing it.

**Reply by ReagentX** [reply]
Can you post the solution you used so that others may read it? I am guessing you just needed to install the [macOS Command Line Tools](https://developer.apple.com/library/archive/technotes/tn2339/_index.html#//apple_ref/doc/uid/DTS40014588-CH1-DOWNLOADING_COMMAND_LINE_TOOLS_IS_NOT_AVAILABLE_IN_XCODE_FOR_MACOS_10_9__HOW_CAN_I_INSTALL_THEM_ON_MY_MACHINE_) with `xcode-select --install`.
